examples/pytorch/image-classification/run_image_classification_no_trainer.py

```diff
@@ -467,17 +473,20 @@ def collate_fn(examples):
                 if resume_step is not None and step < resume_step:
                     completed_steps += 1
                     continue
-            outputs = model(**batch)
-            loss = outputs.loss
-            # We keep track of the loss at each epoch
-            if args.with_tracking:
-                total_loss += loss.detach().float()
-            loss = loss / args.gradient_accumulation_steps
-            accelerator.backward(loss)
-            if step % args.gradient_accumulation_steps == 0 or step == len(train_dataloader) - 1:
+
+            with accelerator.accumulate(model):
+                outputs = model(**batch)
+                loss = outputs.loss
+                # We keep track of the loss at each epoch
+                if args.with_tracking:
+                    total_loss += loss.detach().float()
+
+                accelerator.backward(loss)
                 optimizer.step()
                 lr_scheduler.step()
                 optimizer.zero_grad()
+
+            if accelerator.sync_gradients:
```

Comment:
I think it's going to be easier to just always update, at least for now.

Reply:
If we always update and we have gradient_accumulation_steps > 1, the progress bar will act wrong i.e. increment fast and then stop while still performing updates.

This is because if max_train_steps is not given as an argument, its calculated based on dividing with gradient_accumulation_steps.

Reply:
Excellent point. I think the proper solution here would be a small wrapper around the progress bar (on top of the new one Accelerate just added) that is tied to the `GradientState` similar to how `optimizer.step()` works, where if the gradients should be synchronized then we assume a step has been performed and the progress bar moves. Otherwise it does nothing. 

But for now what you have here does look right to me.